### DISCUSSION TITLE
Mark files that have been deleted on the server.

### DISCUSSION THREAD

**Discussion by YamatoRyou** [element-hq/element-meta#640] [Ideas]
#### What would you like to do?
If a file is deleted on the server, but still exists in the room's stream, use a flag to inform the user that the file no longer exists on the server.
When the user tries to download the file, somehow inform the user that the file cannot be downloaded, instead of popping up a select path dialog.
![{843A582B-5758-7C8A-2188-7E0C3D8EA33A}](https://user-images.githubusercontent.com/48104960/190450525-9979e9f6-91e1-49f7-a0fe-2a616a203eb2.png)

#### Why would you like to do it?
The current handling of such files is somewhat ugly: even if the file has been deleted by the server, the user can still download the file forcibly, and the file just contains JSON like this:
`{"errcode":"M_NOT_FOUND","error":"Not found [b'*******************.com', b'EfbkyHNJEXrwRFAzxHjnbIhJ']"}`

![{9000A931-BCA6-79B9-7AA6-089E58A96C6F}](https://user-images.githubusercontent.com/48104960/190449313-ead7da1d-58c6-4e18-a133-e390058f559a.png)

#### How would you like to achieve it?
n/a

### Have you considered any alternatives?

Use another external file transfer service, such as Jirafeau or PsiTrasnfer. An external file transfer service can solve the following problems:
1. Synapse fails to upload large files inexplicably (https://github.com/matrix-org/synapse/issues/12937);
2. An additional note can be attached to inform the user of the expiration date of the document;
3. Batch download;
4. vector-im/element-web#22369

### Additional context

n/a

**Comment by t3chguy** [comment]
This would require a HEAD request to each and every bit of media to check if it is still present, which would be a significant privacy leak as then your server would know exactly where your client is scrolling

**Reply by YamatoRyou** [reply]
Let users make choices, such as adding an experimental feature:
a) always automatically check if the file exists (automatically, but like you said the server might know the context I'm looking at);
b) only check files in the chat context currently displayed in the window (same as above, but can reduce the number of requests to a certain extent);
c) Only start checking when user clicks on a file (minimum number of requests, Element for Android already implemented to some extent).

As for "the server is trying to spy on the context I'm viewing", would downloading the file from the "Files" tab avoid this issue? I'm using a self-hosted server though, and I'm the owner, so this issue is not for me serious.
